packages/sdk/src/interfaces/text-to-image.ts

```diff
@@ -0,0 +1,80 @@
+import { InterfaceDeclaration } from '../integration/definition'
+import z from '../zui'
+
+const ImageModelRefSchema = z.object({
+  id: z.string(),
+})
+
+const ImageModelSchema = z.object({
+  id: z.string(),
+  name: z.string(),
+  costPerImage: z.number().describe('Cost per image generation, in U.S. dollars'),
+  sizes: z.array(z.string()).describe('Available image sizes'),
+  defaultSize: z.string().describe('Default image size generated by model'),
+})
+
+const ImageGenerationParamsSchema = z
+  .object({
+    id: z.string(), // TODO: remove
+  })
+  .describe('Model-specific parameters for image generation')
+
+const GenerateImageInputSchema = <TModelRef extends z.ZodSchema, TParams extends z.ZodSchema>(
+  imageModelRefSchema: TModelRef,
```

Comment:
**chore:** its confusing but
1. these are not **typescript type arguments**
2. they are **Botpress** type arguments,
3. but they are variables of **known typescript types** (ZodRef)

So this code should look like:
```ts
const GenerateImageInputSchema = (imageModelRefSchema: z.ZodRef, paramsSchema: z.ZodRef) => { ... }
```